samples/azureeventsourcelistener-eventhubs-logging/Program.cs

```diff
@@ -0,0 +1,140 @@
+            }
+
+            // If there was no instrumentation Key provided, then prompt for one.
+            while (string.IsNullOrEmpty(instrumentationKey))
+            {
+                Console.Write("Please provide the Event Hub Name that you'd like to use and then press Enter: ");
+                instrumentationKey = Console.ReadLine().Trim();
+                Console.WriteLine();
+            }
+
+            //Create ApplicationInsights Client.
+            var telemetryClient = new TelemetryClient(new TelemetryConfiguration(instrumentationKey));
+            
+            //Create EventHub Producer Client.
+            var producerClient = new EventHubProducerClient(connectionString, eventHubName);
+
+            try
+            {
+                // Setup a listener to monitor logged events.
+                var listener = new AzureEventSourceListener((args, message) =>
+                {
+                    // Add condition to manipulate only events related to EventHubs.
+                    if (args.EventSource.Name.StartsWith("Azure-Messaging-EventHubs"))
+                    {
+                        var properties = GetPropertiesDictionnary(args.PayloadNames, args.Payload);
+                        var level = args.Level.GetSeverityLevel();
+                        var traceMessage = string.Format(args.Message, args.Payload.ToArray());
+                        telemetryClient.TrackTrace($"Sample_AzureEventSourceListener | {traceMessage}", level, properties);
+                        telemetryClient.Flush();
+                    }
+                }, EventLevel.LogAlways);
+
+                // Creating a producer client using its default set of options.
+                // We will publish three small batch events.
+
+                Console.WriteLine("Preparing event batch...");
+                using EventDataBatch eventBatch = await producerClient.CreateBatchAsync();
+                // Add events to the batch. An event is a represented by a collection of bytes and metadata.
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Hello, Event Hubs!")));
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("The middle event is this one")));
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Goodbye, Event Hubs!")));
+                await producerClient.SendAsync(eventBatch);
+                Console.WriteLine("The event batch has been published.");
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"An exception of type { ex.GetType().Name } occurred.  Message:{ Environment.NewLine }\t{ ex.Message }");
+            }
+            finally
+            {
+                await producerClient.CloseAsync();
+                Console.WriteLine();
+                Console.WriteLine();
+                Console.WriteLine("Exiting...");
+            }
+        }
+
+        /// <summary>
```

Comment:
nit:  The spacing and indentation for this block differs from the class.   Either format is fine so long as we're consistent within the project.